test/e2e/snaps/test-snap-manageAccount.spec.js

```diff
@@ -1,106 +1,452 @@
         ganacheOptions,
         failOnConsoleError: false,
         title: this.test.title,
       },
       async ({ driver }) => {
-        await driver.navigate();
-
-        // enter pw into extension
-        await driver.fill('#password', 'correct horse battery staple');
-        await driver.press('#password', driver.Key.ENTER);
-
-        // navigate to test snaps page and connect
-        await driver.openNewPage(TEST_SNAPS_SIMPLE_KEYRING_WEBSITE_URL);
-        await driver.delay(1000);
-        const connectButton = await driver.findElement('#connectButton');
-        await driver.scrollToElement(connectButton);
-        await driver.delay(1000);
-        await driver.clickElement('#connectButton');
-        await driver.delay(500);
-
-        // switch to metamask extension and click connect
-        const windowHandles = await driver.waitUntilXWindowHandles(
-          3,
-          1000,
-          10000,
-        );
-        await driver.switchToWindowWithTitle(
-          'MetaMask Notification',
-          windowHandles,
-        );
-        await driver.clickElement({
-          text: 'Connect',
-          tag: 'button',
-        });
-
-        await driver.clickElementSafe('[data-testid="snap-install-scroll"]');
-
-        await driver.waitForSelector({ text: 'Install' });
-
-        await driver.clickElement({
-          text: 'Install',
-          tag: 'button',
-        });
-
-        await driver.waitForSelector({ text: 'OK' });
-
-        await driver.clickElement({
-          text: 'OK',
-          tag: 'button',
-        });
-
-        await driver.switchToWindowWithTitle(
-          'SSK - Snap Simple Keyring',
-          windowHandles,
-        );
-
-        // check the dapp connection status
-        await driver.waitForSelector({
-          css: '#snapConnected',
-          text: 'Connected',
-        });
-
-        // create new account on dapp
-        await driver.clickElement({
-          text: 'Create account',
-          tag: 'div',
-        });
-
-        await driver.clickElement({
-          text: 'Create Account',
-          tag: 'button',
-        });
-
-        await driver.delay(1000);
-
-        // switch to metamask extension
-        await driver.switchToWindowWithTitle('MetaMask', windowHandles);
-
-        // click on accounts
-        await driver.clickElement('[data-testid="account-menu-icon"]');
-
-        const label = await driver.findElement('.mm-tag');
-        assert.strictEqual(await label.getText(), 'Snaps');
+        const flowType = 'approve';
+        const isAsyncFlow = flowType !== 'sync';
+
+        await installSnapSimpleKeyring(driver, isAsyncFlow);
+
+        const newPublicKey = await makeNewAccountAndSwitch(driver);
+
+        // open the Test Dapp and connect Account 2 to it
+        await connectAccountToTestDapp(driver);
+
+        // do #signTypedDataV3
+        await signData(driver, '#signTypedDataV3', newPublicKey, flowType);
```

Comment:
Do we need an assertion here to verify that this worked and to confirm that the signing address is verified against the public address used?